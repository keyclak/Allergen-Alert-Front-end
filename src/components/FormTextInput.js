import React, { useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';
import { Colors, StyleConstants, Styles } from '../style';

export default function FormTextInput({error, ...params}) {
    const [focused, setFocused] = useState(false);

    function onFocus() {
        setFocused(true);
    }

    function onBlur() {
        setFocused(false);
    }

    return (
        <TextInput 
            onFocus={onFocus}
            onBlur={onBlur}
            {...params}
            placeholderTextColor={error ? Colors.Red[5] : Colors.Gray[6]}
            style={[Styles.formInput, error && Styles.formInputError, focused && Styles.formInputFocus]}/>
    )
}