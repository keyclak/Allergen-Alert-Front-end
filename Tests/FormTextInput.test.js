import renderer from 'react-test-renderer';
import React from 'react';
import MockDate from 'mockdate'
import FormTextInput from '../src/components/FormTextInput';

const FRAME_TIME = 10

const advanceOneFrame = () => {
  const now = Date.now();
  MockDate.set(new Date(now + FRAME_TIME));
  jest.advanceTimersByTime(FRAME_TIME);
}

const timeTravel = (msToAdvance = FRAME_TIME) => {
  const numberOfFramesToRun = msToAdvance / FRAME_TIME;
  let framesElapsed = 0;

  while (framesElapsed < numberOfFramesToRun) {
    advanceOneFrame();
    framesElapsed++;
  }
}
global.requestAnimationFrame = cb => {
  setTimeout(cb, FRAME_TIME);
}

export const setupTimeTravel = () => {
    MockDate.set(0);
    jest.useFakeTimers();
}

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('@expo/vector-icons', () => ({
    MaterialIcons: 'Icon'
}));

beforeEach(setupTimeTravel);

describe("FormTextInputTests", () => {
    test("Renders correctly", async () => {
        const tree = renderer.create(
            <FormTextInput error={jest.fn()} />
        ).toJSON(); 

        timeTravel(500);

        expect(tree).toMatchSnapshot();
    })
})