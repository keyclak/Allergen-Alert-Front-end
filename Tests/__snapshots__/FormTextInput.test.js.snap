// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInputTests Renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  onBlur={[Function]}
  onFocus={[Function]}
  placeholderTextColor="#DA3542"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "backgroundColor": "#e9ecef",
        "borderColor": "#868e96",
        "borderRadius": 15,
        "borderWidth": 0,
        "color": "#495057",
        "fontSize": 16,
        "height": 50,
        "marginBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
      },
      Object {
        "backgroundColor": "#FEC9B6",
        "borderColor": "#DA3542",
      },
      false,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
